### ISSUE TITLE
How do I perform validation?

### ISSUE THREAD

**Issue by Igal20** [facebookresearch/maskrcnn-benchmark#171]
## ❓ Questions and Help
Hello, 
Once in a while I would like to perform test on validation set, what is right way to do so ?
THanks

**Comment by fmassa** [comment]
Hi,

You could add a condition in [here](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/engine/trainer.py#L105) so that it calls the [inference function](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/tools/train_net.py#L95) to validate on a dataset.
You might need to pass a few extra arguments around, but it should be fairly straightforward.

But given that we are doing training on multiple GPUs, if you want to perform testing on multiple GPUs as well you might need to unwrap the `model` from the `DistributedDataParallel`, add a `synchronize` call in the end of the inference and wrap the model back in `DistributedDataParallel` just before the training starts again.

Let me know if you have further questions

**Comment by JoyHuYY1412** [comment]
> unwrap the `model` from the `DistributedDataParallel`, add a `synchronize` call

Sorry but how to unwrap the model?

Like what should I change in this code:
`if iteration % checkpoint_period == 0 or iteration == max_iter:
         model.eval()
          iou_types = ("bbox",)
            if cfg.MODEL.MASK_ON:
                iou_types = iou_types + ("segm",)
                    inference(
                        model,
                        val_data_loader,
                        iou_types=iou_types,
                        box_only=cfg.MODEL.RPN_ONLY,
                        device=cfg.MODEL.DEVICE,
                        expected_results=cfg.TEST.EXPECTED_RESULTS,
                        expected_results_sigma_tol=cfg.TEST.EXPECTED_RESULTS_SIGMA_TOL,
                        verbose=False
                    )

                model.train()  # reset training flag
            synchronize()
`

**Comment by fmassa** [comment]
@JoyHuYY1412 yes, this makes sense, but I won't guarantee that it will work 100% in all cases.

**Comment by JoyHuYY1412** [comment]
> @JoyHuYY1412 yes, this makes sense, but I won't guarantee that it will work 100% in all cases.

It does not work with multigpu. I don't how to tell it just use one GPU in validation
